### PR TITLE
Add end_time field

### PR DESCRIPTION
**Pull request by ela-kotulska-frequenz**

### CODE REVIEW

File: proto/frequenz/api/dispatch/v1/dispatch.proto [hunk 1]
```diff
@@ -147,6 +147,11 @@ message DispatchDetail {
 
   // UTC Timestamp of the last update to the order.
   google.protobuf.Timestamp modification_time = 4;
+
+  // UTC Timestamp when the dispatch will stop working.
+  // Will be calculated internally based on the given: start_time,
+  // duration and RecurenceRule
+  google.protobuf.Timestamp estimated_end_time = 5;
```
Comment (Marenz) [inline proto/frequenz/api/dispatch/v1/dispatch.proto]: I wonder if we should just call it "end_time", it's not really an estimation but a precise calculation, no?

Comment (ela-kotulska-frequenz) [reply proto/frequenz/api/dispatch/v1/dispatch.proto]: Good point, done :)

**Review by Marenz — APPROVED** [review]

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ In this release, we have made some changes to the API to improve the user experi
 * Add parameters for pagination and sorting.
 * Documentation of valid values for `count` and `interval` fields was added.
 * Extended filter parameters for recurrence and end time.
+* Add new field end\_time to the DispatchDetail.
```
Comment (llucax) [inline RELEASE_NOTES.md:25]: Too late, but for next time I suggest:
```markdown
* Add new field `end_time` to the `DispatchDetail`.
```

**Comment by llucax** [comment]
Is there an issue about this? Why are we adding it to the protocol if it is calculated from other fields and clients already need to do this anyway?

File: RELEASE_NOTES.md (hunk 2 above)
Comment (ela-kotulska-frequenz) [reply RELEASE_NOTES.md:25]: Sure, thanks!

**Comment by ela-kotulska-frequenz** [comment]
@leandro-lucarella-frequenz 

Issue: https://github.com/frequenz-io/frequenz-service-dispatch/issues/82

Shortly: 
1) We would like to have an option to sort/filter dispatches by end time.
2) Client doesn't need to calculate the end_time.  End_criteria for reoccurring dispatches is either:
   a) optional - infinite dispatch
   b) count (how many times to execute) 
   c) until - max end time  (not accurate)

In b) and c) we need to calculate accurate end time and save as `end_time` in the database.

**Comment by llucax** [comment]
Thanks! BTW, I'm now using the GitHub handle @llucax, I only use the `frequenz` one as admin, so in my every-day work I don't have admin access.

**Pull request merged**